nengo/spa/actions.py

```diff
@@ -154,7 +154,11 @@ class Actions(object):
     def __init__(self, *args, **kwargs):
         self.actions = None
         self.args = args
-        self.kwargs = kwargs
+        self.kwargs = OrderedDict(kwargs)
```

Comment:
Ah, looking at the last commit, I see now. One thing worth noting is it doesn't have as much to do with the Python version as it has to do with the nature of undefined behaviour. You could reorganize the code slightly (maybe change the name of the keyword arg) and then suddenly you could get the flip behaviour of it working in the opposite version but not the other (or both, or neither), or even failing on different machines.

At any rate, the comment about it being misleading is still a concern. To be clear about what I mean...

```
actions.add(a=x, b=y)
```

could have the order `('a', 'b')` in some cases, and the order `('b', 'a')` in other cases, due to the undefined ordering of `**kwargs`. The order will only be preserved under the condition that `len(kwargs) <= 1`.

IMO, making it a list again might be the way to go, to avoid having a user rely on the behaviour only to inevitably encounter a weird issue like the above example. To make the tests work, you could do something like sort the actions in the test?